src/Csv.cpp

```diff
@@ -2,10 +2,16 @@
 
 #include <QDebug>
 
-Csv::Csv(QString delimiter, QString quote)
+Csv::Csv(QString delimiter, QString quote, bool includeHeader, bool quoteStringColumns, QLocale locale, QHash<QString, QString> formatOverrides) :
+    m_delimiter(delimiter),
+    m_quote(quote),
+    m_includeHeader(includeHeader),
+    m_locale(locale),
+    m_quoteStringColumns(quoteStringColumns)
 {
-    m_delimiter = delimiter;
-    m_quote = quote;
+    m_dateFormat = formatOverrides.contains("dateFormat") ? formatOverrides["dateFormat"] : locale.dateFormat(QLocale::ShortFormat);
```

Comment:
It might be worth separating concerns here. You could have a new class with one method that takes QVariant and produces QString based type, locale, and settings.

Reply:
not sure what you mean. like class ExportParameters with various members?

Reply:
ok, i'll take this class prototype and try to do something.